lib/client/doc.js

```diff
@@ -960,6 +960,22 @@ Doc.prototype._opAcknowledged = function(message) {
     return this.fetch();
   }
 
+  if (message.fu) {
+    for (var i = 0; i < message.fu.length; i++) {
+      var fixupOp = message.fu[i];
+      try {
+        this._otApply(fixupOp, false);
+      } catch (error) {
+        return this._hardRollback(error);
+      }
+
+      for (var j = 0; j < this.pendingOps.length; j++) {
+        var transformErr = transformX(this.pendingOps[i], fixupOp);
+        if (transformErr) return this._hardRollback(transformErr);
+      }
```

Comment:
The order here of apply -> transform pending is the opposite of what happens in `_handleOp` for a remotely issued op:
https://github.com/share/sharedb/blob/c5b90c964a2e5ff10a61ab2fd4dcf3987f3147c6/lib/client/doc.js#L369-L384

We had some discussion around whether the order matters, Alec then tried to write a test case where having the "original" order from `_handleOp` might matter, but it's not quite failing with this order here due to json0 being a bit lenient?
https://github.com/share/sharedb/pull/581/files#r1035068235

Conclusion seems to be that we should swap this to match the `_handleOp` order